### ISSUE TITLE
Question about the validation set and model saving logic for survival analysis

### ISSUE THREAD

**Issue by Haojiang-11** [DearCaat/E2E-WSI-ABMILX#4]
Hi authors,

First of all, thank you for your excellent work and for making the code publicly available. I have read your paper with great interest.

I have a question regarding the data splitting procedure described in Appendix B.1. In the paper, you outline your method for **splitting the datasets.** However, when examining the code for the survival prediction task, I noticed that the data splits seem to **only contain training and testing sets, without a dedicated validation set.**

This leads me to two questions:

How were the performance metrics for the survival prediction task obtained without a validation set for model selection?

Could you please clarify the logic for saving the final model checkpoint during training? For instance, was it saved after a fixed number of epochs, or was another method used?

Thank you for your time and for your valuable contribution to the field. I look forward to your clarification.

<img width="1710" height="169" alt="Image" src="https://github.com/user-attachments/assets/5043f0fb-87a5-4fcd-83d2-ebe96b354869" />  

<img width="1805" height="333" alt="Image" src="https://github.com/user-attachments/assets/85d42335-9fce-4ec1-b1ee-b8c932392142" />

**Comment by DearCaat** [comment]
I think this may be the same question. The logic for saving the best checkpoint in the code is designed to use the validation set when it exists for the task; if not, it falls back to using the test set. This approach is mainly adopted due to the limited sample sizes in some tasks, especially for survival prediction (which typically has fewer than 500 samples). For datasets like PANDA with sufficient samples, we use an independent validation set to select the best checkpoint as usual, and finally test the model on the test set.

Thanks for attention. and look forward further discussion.

**Comment by Haojiang-11** [comment]
Got it. So, am I correct in understanding that you save the model that performs best (**e.g., has the minimum loss**) on the test set itself?
